ios/HybridImageFactory.swift

```diff
@@ -21,14 +22,102 @@ class HybridImageFactory: HybridImageFactorySpec {
     guard let url = URL(string: urlString) else {
       throw RuntimeError.error(withMessage: "URL string \"\(urlString)\" is not a valid URL!")
     }
-
+    
     return Promise.async {
       let webImageOptions = options?.toSDWebImageOptions() ?? []
       let uiImage = try await SDWebImageManager.shared.loadImage(with: url, options: webImageOptions)
       return HybridImage(uiImage: uiImage)
     }
   }
   
+  /**
+   * Load Image from URL
+   */
+  func loadFromAssetAsync(assetId: String, options: AssetImageLoadOptions?) throws -> Promise<any HybridImageSpec> {
+    let assets = PHAsset.fetchAssets(
+      withLocalIdentifiers: [assetId],
+      options: nil
+    )
+    guard let asset = assets.firstObject else {
+      throw NSError(
+        domain: "TurboImageView",
+        code: 404,
+        userInfo: [NSLocalizedDescriptionKey: "Asset not found"]
+      )
```

Comment:
This is not a `TurboImageView`, and there's no reason in throwing boxed ObjC errors either.

```suggestion
      throw RuntimeError.error(withMessage: "Asset with ID \(assetId) was not found!")
```